Mathlib/Topology/CWComplex/Basic.lean

```diff
@@ -0,0 +1,682 @@
+/-
+Copyright (c) 2024 Floris van Doorn and Hannah Scholz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn, Hannah Scholz
+-/
+
+import Mathlib.Analysis.InnerProductSpace.Basic
+import Mathlib.Data.Real.StarOrdered
+import Mathlib.Order.CompletePartialOrder
+
+/-!
+# CW-complexes
+
+This file defines (relative) CW-complexes and proofs basic properties about them.
+
+A CW-complex is a topological space that is made by gluing closed disks of different dimensions
+together.
+
+## Main definitions
+* `RelCWComplex C D` : The class of CW-structures on a subspace `C` realtive to `D` of a topological
+  space `X`.
+* `ClasCWComplex C` : An abbreviation for `RelCWComplex C ∅`. The class of CW-structures on a
+  subspace `C` of the topological space `X`.
+* `openCell n i` : An open cell of dimension `n`.
```

Comment:
minor comment: in English, you're supposed to continue in lower-case after a colon. (With very few exceptions.)
(This is different in e.g. German. Yes, this can be confusing.)